src/main/kotlin/no/nav/familie/ef/iverksett/økonomi/konsistens/KonsistensavstemmingService.kt

```diff
@@ -0,0 +1,99 @@
+package no.nav.familie.ef.iverksett.økonomi.konsistens
+
+import no.nav.familie.ef.iverksett.infrastruktur.transformer.toDomain
+import no.nav.familie.ef.iverksett.iverksetting.domene.TilkjentYtelse
+import no.nav.familie.ef.iverksett.iverksetting.tilstand.TilstandRepository
+import no.nav.familie.ef.iverksett.økonomi.OppdragClient
+import no.nav.familie.ef.iverksett.økonomi.tilKlassifisering
+import no.nav.familie.ef.iverksett.økonomi.utbetalingsoppdrag.lagPeriodeFraAndel
+import no.nav.familie.kontrakter.ef.felles.StønadType
+import no.nav.familie.kontrakter.felles.oppdrag.KonsistensavstemmingUtbetalingsoppdrag
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsoppdrag
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.UUID
+
+@Service
+class KonsistensavstemmingService(
+        private val oppdragKlient: OppdragClient,
+        private val tilstandRepository: TilstandRepository
+) {
+
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    fun sendKonsistensavstemming(konsistensavstemmingDto: KonsistensavstemmingDto) {
+        try {
+            val utbetalingsoppdrag = lagUtbetalingsoppdragForKonsistensavstemming(konsistensavstemmingDto)
+            sendKonsistensavstemming(utbetalingsoppdrag, konsistensavstemmingDto.stønadType)
+        } catch (feil: Throwable) {
+            throw Exception("Sending av utbetalingsoppdrag til konsistensavtemming feilet", feil)
+        }
+    }
+
+    private fun lagUtbetalingsoppdragForKonsistensavstemming(konsistensavstemmingDto: KonsistensavstemmingDto)
+            : List<Utbetalingsoppdrag> {
+        val stønadType = konsistensavstemmingDto.stønadType
+        val tilkjentYtelsePerBehandlingId = konsistensavstemmingDto.tilkjenteYtelser.associateBy { it.behandlingId }
+        val behandlingIdTilkjentYtelseForUtbetalingMap = tilstandRepository.hentTilkjentYtelse(tilkjentYtelsePerBehandlingId.keys)
+
+        return behandlingIdTilkjentYtelseForUtbetalingMap.map { (behandlingId, tilkjentYtelse) ->
+            genererUtbetalingsoppdrag(tilkjentYtelsePerBehandlingId[behandlingId]!!,
+                                      tilkjentYtelse,
+                                      behandlingId,
+                                      stønadType)
+        }
+    }
+
+    private fun genererUtbetalingsoppdrag(konsistensavstemmingTilkjentYtelseDto: KonsistensavstemmingTilkjentYtelseDto,
```

Comment:
Det som skjer i denne er att vi henter opp andelene fra databasen for å finne periodeId og forrigePeriodeId.
Det gjøres en match som sjekker att alle beløp/perioder er direkt, hvis noen mangler kastes en feil